packages/core-data/src/entities.ts

```diff
@@ -460,6 +461,9 @@ async function loadPostTypeEntities() {
 				( isTemplate ? startCase( record.slug ) : String( record.id ) ),
 			__unstablePrePersist: isTemplate ? undefined : prePersistPostType,
 			__unstable_rest_base: postType.rest_base,
+			...( name === `wp_navigation` && {
+				key: 'slug',
+			} ),
```

Comment:
This forces the Navigation posts to be keyed by slug in the state. This ensures that `getEntityRecord` can retrieve them by `slug`, otherwise it defaults to `ID` which won't work.